compactor/periodic.go

```diff
@@ -86,6 +86,10 @@ func (t *Periodic) Run() {
 			}
 			plog.Noticef("Starting auto-compaction at revision %d (retention: %v)", rev, t.period)
 			_, err := t.c.Compact(t.ctx, &pb.CompactionRequest{Revision: rev})
+
+			// update the last compaction time
+			last = clock.Now()
```

Comment:
why adding here? should we record last compaction time only when it succeeds?
put it under ?
```go
if err == nil || err == mvcc.ErrCompacted {
    t.revs = remaining
    last = clock.Now()
```

Reply:
Think it makes sense to reset `last` for every `Compact` call, whether it's failed or not.
Otherwise, the interval would be 1/10 of original period, once `clock.Now().Sub(last) >= t.period`.

Reply:
@gyuho the bug was introduced here https://github.com/coreos/etcd/pull/8563#discussion-diff-174682081L90 
as you can see, the original code sets `last` only when compaction succeeds.

Reply:
Yeah saw that. We can just reset `last` only when it succeeds, and maybe change this later in refactoring.

Reply:
Actually, when I write this fix, I think it a while.I think we should update the `last` whether failed or not. when it failed, we don't want to retry faster either(10X faster), otherwise you should write a doc to tell the user.

Reply:
@WIZARD-CXY Yeah, but we want to keep old behavior. And we can improve afterwords.
Let's fix the broken behavior first (to be backported to 3.3).

Reply:
I think it is ok we should do this right now as @xiang90 the original author says "looks reasonable to me" too. I think we don't need to wait some time IMHO

Reply:
I am fine with that too. Can you make those changes in separate commits then?
That would make my backport easier.

Reply:
after second thought, we'd better only update the last when the compact succeeds. Or we have to wait for another compaction duration rather than `checkCompactInterval`. It also makes the logging at line 98 wrong. 

let us fix this as what @fanminshi suggested.

Reply:
My concern is that I don't want to retry faster(10x faster is too fast) less safe when compact fail.

Reply:
@WIZARD-CXY if you have a compaction duration, which is 10hours. you 100% do not want to retry after another 10 hours. it means that your etcd db size will likely be doubled.

Reply:
@xiang90 the thing is compaction duration is 10min in our case, not 10hr

Reply:
Actually it is all about different user cases. For fixing this bug first, I agree to change it to update the `last` only when compaction goes successfully.